src/discordcr/mappings/channel.cr

```diff
@@ -17,6 +17,63 @@ module Discord
       embeds: Array(Embed),
       pinned: {type: Bool, nilable: true}
     )
+
+    # A mention contained within a message, constructed by its type,
+    # the snowflake, and it's name if it is an emoji.
+    alias Mention = Tuple(MentionType, UInt64?, String?)
```

Comment:
I'd return just an array of all mention types too. It's easy enough to use `mentions.select(&.type == User)`, and for typical messages (less than 5 mentions) it'll be faster too. Worse case is only `O(n)` as well.